src/Swashbuckle.AspNetCore.Cli/Program.cs

```diff
@@ -156,9 +126,103 @@ public static int Main(string[] args)
                 });
             });
 
+            // > dotnet swagger list
+            runner.SubCommand("list", "retrieves the list of Swagger document names from a startup assembly", c =>
+            {
+                c.Argument("startupassembly", "relative path to the application's startup assembly");
+                c.Option("--output", "relative path where the document names will be output, defaults to stdout");
+                c.OnRun((namedArgs) =>
+                {
+                    string subProcessCommandLine = PrepareCommandLine(args, namedArgs);
+
+                    var subProcess = Process.Start("dotnet", subProcessCommandLine);
+
+                    subProcess.WaitForExit();
+                    return subProcess.ExitCode;
+                });
+            });
+
+            // > dotnet swagger _list ... (* should only be invoked via "dotnet exec")
+            runner.SubCommand("_list", "", c =>
+            {
+                c.Argument("startupassembly", "");
+                c.Option("--output", "");
+                c.OnRun((namedArgs) =>
+                {
+                    SetupAndRetrieveSwaggerProviderAndOptions(namedArgs, out var swaggerProvider, out var swaggerOptions);
+                    var docMetaProvider = swaggerProvider as ISwaggerDocumentMetadataProvider;
+                    var docNames = docMetaProvider.GetDocumentNames();
+                    var outputPath = namedArgs.TryGetValue("--output", out var arg1)
+                        ? Path.Combine(Directory.GetCurrentDirectory(), arg1)
+                        : null;
+                    bool outputViaConsole = outputPath == null;
+                    if (!string.IsNullOrEmpty(outputPath))
+                    {
+                        string directoryPath = Path.GetDirectoryName(outputPath);
+                        if (!string.IsNullOrEmpty(directoryPath) && !Directory.Exists(directoryPath))
+                        {
+                            Directory.CreateDirectory(directoryPath);
+                        }
+                    }
+
+
+                    using Stream stream = outputViaConsole ? Console.OpenStandardOutput() : File.Create(outputPath);
+                    using StreamWriter writer = new StreamWriter(stream, outputViaConsole ? Console.OutputEncoding : Encoding.UTF8);
+
+                    writer.WriteLine($"Swagger Document Names (Surrounded by single quotes):");
+                    foreach (var name in docNames)
+                    {
+                        writer.WriteLine($"'{name}'");
```

Comment:
Why quotes, and if quotes, why not double-quotes?

Reply:
Well, since the document names can be arbitrary values, it's a good idea to surround them with quotes. 
And, I'll tend to default to picking the opposite quote character for output wrapping than the string literal I'm using by default from too much JavaScript code writing.. :) That said... I've switched this to double quotes.